### DISCUSSION TITLE
hosting commercial fonts

### DISCUSSION THREAD

**Discussion by freeman-lab** [vercel/next.js#11899] [Help]
Hi all! Really loving `next.js`. Trying to solve a problem with including commercial fonts on my site. I have the font files (with a license to use on the web) and am successfully using [`next-fonts`](https://github.com/rohanray/next-fonts) to use them on the site. But per the terms of my license, I'm requested to prevent direct access to the font files. Is there any way to protect the contents of `public/fonts` this way with `next`, maybe through an appropriate custom CORS configuration? Or should I explore a third-party CDN and whitelist my domain(s)? Any help would be appreciated!

**Comment by Timer** [comment]
The default CORS configuration of your application should prevent other sites from accessing the file.

**Reply by freeman-lab** [reply]
Thanks! That's helpful to know. But it would be still possible to identify the font file URL via network resources and download it directly right? Is there any custom config that would prevent that?

**Reply by Timer** [reply]
@freeman-lab it's not possible to paywall your font file specifically unless your entire site requires authentication.

**Reply by timneutkens** [reply]
> Thanks! That's helpful to know. But it would be still possible to identify the font file URL via network resources and download it directly right? Is there any custom config that would prevent that?

Just in general that would mean just not using the font on the site so that it's not sent to users. Users will always be able to download anything you put on a website.

If the terms of the font are really like that it sounds unlikely that it's meant to be used as a web font and is meant for eg printing / designs only.
